src/components/VoiceSearch/VoiceSearch.tsx

```diff
@@ -0,0 +1,95 @@
+import React from 'preact-compat';
+import PropTypes from 'prop-types';
+import Template from '../Template/Template';
+
+import {
+  VoiceSearchCSSClasses,
+  VoiceSearchTemplates,
+  VoiceListeningState,
+} from '../../widgets/voice-search/voice-search';
+
+type VoiceSearchProps = {
+  cssClasses: VoiceSearchCSSClasses;
+  isSupportedBrowser: () => boolean;
+  isListening: () => boolean;
+  toggleListening: (searchAsYouSpeak: boolean) => void;
+  voiceListeningState: VoiceListeningState;
+  searchAsYouSpeak: boolean;
+  templates: VoiceSearchTemplates;
+};
+
+const VoiceSearch = ({
+  cssClasses,
+  isSupportedBrowser,
+  isListening,
+  toggleListening,
+  voiceListeningState,
+  searchAsYouSpeak,
+  templates,
+}: VoiceSearchProps) => {
+  const handleClick = (event: any) => {
+    event.currentTarget.blur();
+    toggleListening(searchAsYouSpeak);
+  };
+
+  const listening = isListening();
+  const { status, transcript, isSpeechFinal, errorCode } = voiceListeningState;
+  return (
+    <div className={cssClasses.root}>
+      <Template
+        templateKey="buttonText"
+        rootTagName="button"
+        rootProps={{
+          className: cssClasses.button,
+          type: 'button',
+          title: 'Search by voice',
+          onClick: handleClick,
+          disabled: !isSupportedBrowser(),
+        }}
+        data={{
+          status,
+          errorCode,
+          isListening: listening,
+          transcript,
+          isSpeechFinal,
+        }}
+        templates={templates}
+      />
+      <Template
+        templateKey="status"
+        rootTagName="div"
+        rootProps={{
+          className: cssClasses.status,
+        }}
+        data={{
+          status,
+          errorCode,
+          isListening: listening,
+          transcript,
+          isSpeechFinal,
+        }}
+        templates={templates}
+      />
+    </div>
+  );
+};
+
+VoiceSearch.propTypes = {
```

Comment:
We don't need PropTypes now that we use TypeScript.

Reply:
As far as I know, PropTypes provides runtime type checking, but assuming our users will not use the component directly, so there won't be any case of wrong props delivered to the component, we can remove Proptypes and stick to TypeScript for compile time type checking. Is this correct?

Reply:
Yes – PropTypes don't bring much more than TypeScript in this project since we're the only consumers (our components are not exposed). It's not worth carrying that redundancy for new components.

Reply:
Clear. Thanks!